### PR TITLE
fix(flags): don't let users who can't edit flags access the enable/disable toggle

### LINKED ISSUE #25440: Bug Report: RBAC - User w/out perms for feature flags can view and toggle the "Enabled" state (but state doesn't actually unchange)
## Bug description

A user who is not a member of a group with permissions to view or edit feature flags, can view flags and toggle the `Enabled` state on a flag. (The state change does not actually happen, Refreshing the page shows the state is unchanged.)

## How to reproduce

1. Log in as a user who is not a member of an RBAC group with perms to edit feature flags(e.g. [this user](https://eu.posthog.com/admin/posthog/user/46099/change/))
2. View a feature flag on the prod project, e.g. [this test flag](https://eu.posthog.com/project/7169/feature_flags/34561)
3. Note that the `Edit` button is greyed-out, as expected.
4. Click the toggle for `Enabled`, then click the `Confirm` button in the confirmation dialog
5. Note that the `Enabled` toggle appears to be toggled off.
6. Navigate to a different page, then return to the feature flag page (or just reload the feature flag page) and note that the flag is actually still `Enabled`.

The expectation is that the user should not be able to view the flag (not in a group with view access) and should not seem to be able to toggle the `Enabled` state, because the user is not in an RBAC group with permissions to view or edit the flag.  

Screenshots showing the user, from step 1 above, is not currently in either group:

CleanShot 2024-10-07 at 10 30 27@2x

CleanShot 2024-10-07 at 10 30 35@2x

CleanShot 2024-10-07 at 10 30 40@2x

## Additional context
From: https://posthoghelp.zendesk.com/agent/tickets/18533

### Debug info

```shell
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/019267ff-4126-7ea1-9084-6ba781d7cb7f?t=980
Admin: http://go/adminOrgEU/01896da5-1508-0000-8376-013b50e81177 (project ID 7169)
Sentry: http://go/sentryEU/7169
```

### PR DESCRIPTION
**Pull request by dmarticus**
## Problem

Closes https://github.com/PostHog/posthog/issues/25440

## Changes

Video demo of new behavior: https://www.loom.com/share/554b5e76761843fc92967054dfa58324

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

 Manual UI testing

### CODE REVIEW

File: cypress.e2e.config.ts [hunk 1]
```diff
@@ -117,5 +125,6 @@ export default defineConfig({
         },
         baseUrl: 'http://localhost:8000',
         specPattern: 'cypress/e2e/**/*.{js,jsx,ts,tsx}',
+        chromeWebSecurity: false,
```
Comment (dmarticus) [inline cypress.e2e.config.ts:128]: gamely attempting to get CI to pass, followed what Georgiy did [here](https://github.com/PostHog/posthog/pull/25458)

File: frontend/src/scenes/feature-flags/FeatureFlag.tsx [hunk 2]
```diff
@@ -783,6 +783,12 @@ function FeatureFlagRollout({ readOnly }: { readOnly?: boolean }): JSX.Element {
                                 })
                             }}
                             label="Enabled"
+                            disabled={!featureFlag.can_edit}
+                            disabledReason={
```
Comment (Twixes) [inline frontend/src/scenes/feature-flags/FeatureFlag.tsx:787]: You only need `disabledReason` (ideally the props type would reflect that it's either one or the other)
```suggestion
                            disabledReason={
```

File: frontend/src/scenes/feature-flags/FeatureFlag.tsx [hunk 3]
```diff
@@ -783,6 +783,12 @@ function FeatureFlagRollout({ readOnly }: { readOnly?: boolean }): JSX.Element {
...
+                                !featureFlag.can_edit
+                                    ? "You have only 'View' access for this feature flag. To make changes, please contact the flag's creator."
```
Comment (Twixes) [inline frontend/src/scenes/feature-flags/FeatureFlag.tsx:789]: Subjective suggestion for slightly cleaner copy:
```suggestion
                                    ? "You only have view access to this feature flag. To make changes, contact the flag's creator."
```

**Review by Twixes — APPROVED** [review]
Only thing to fix before merging is the redundant `disabled` prop, but here's my stamp :)

**Pull request merged**